### ISSUE TITLE
Useful error message for PKIX path building failure with standard switches

### ISSUE THREAD

**Issue by monolied** [gradle/gradle#28977]
### Expected Behavior

I'd like to, with or without reasonable command line switches, receive a useful error message in case of PKIX path building failure. 

### Current Behavior (optional)

I only get something according to
```
* What went wrong:
Plugin [id: 'org.springframework.boot', version: '3.2.5'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (could not resolve plugin artifact 'org.springframework.boot:org.springframework.boot.gradle.plugin:3.2.5')
  Searched in the following repositories:
    Gradle Central Plugin Repository
...
```

### Context

We are, for security reasons, required to only provide the necessary subset of CA certificates. Because of that, we provide gradle with a custom keystore. When an external plugin or library is required, the download therefore sometimes fails, but the error message does not reflect the cause of failure.
Using switches like the recommended `--info`, `--debug`, `--stacktrace` or `--full-stacktrace` deliver more info, but not the one about the certificate error. The only switch providing the error seems to be `--scan`, but that interrupts the build process for the scan upload and [should not be used in our situation](https://discuss.gradle.org/t/automatically-activate-build-scan-to-an-email/42381).
`gradlew` also delivers the error (and I wouldn't know why there is or should be a difference), but our developers do not use it.
The issue is reproducible by providing a build with an empty or incomplete keystore and using the `--refresh-dependencies` switch, e.g. like so:
 `/opt/gradle/gradle-8.6/bin/gradle -V --refresh-dependencies -Djavax.net.ssl.trustStore=/tmp/trust.jks build`
Add switches as you like. If I have overlooked a switch besides `--scan` producing the desired result, I'd also be glad to hear about it.

**Comment by ov7a** [comment]
You can debug the connection using `-Djavax.net.debug=all`:
```
gradle --refresh-dependencies -Djavax.net.ssl.trustStore=./dummy.jks -Djavax.net.debug=all build
```
It will print stacktraces, handshakes, and errors.

**Comment by monolied** [comment]
Hm.

```
gradle -V --refresh-dependencies -Djavax.net.debug=all -Djavax.net.ssl.trustStore=./dummy.jks build
------------------------------------------------------------
Gradle 8.6
------------------------------------------------------------

Build time:   2024-02-02 16:47:16 UTC
Revision:     d55c486870a0dc6f6278f53d21381396d0741c6e

Kotlin:       1.9.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          17.0.11 (Eclipse Adoptium 17.0.11+9)
OS:           Linux 6.1.0-20-amd64 amd64


FAILURE: Build failed with an exception.

* Where:
Build file '/daten/ci/jenkins/workspace/weBBservice-Dev/webbservice/build.gradle' line: 4

* What went wrong:
Plugin [id: 'org.springframework.boot', version: '3.2.5'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (could not resolve plugin artifact 'org.springframework.boot:org.springframework.boot.gradle.plugin:3.2.5')
  Searched in the following repositories:
    Gradle Central Plugin Repository

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 540ms

```

**Comment by ov7a** [comment]
@monolied make sure that this is not a reused result from cache.
The simplest way to do this is to remove `./build` directories and start with a clean gradle home (`export GRADLE_USER_HOME=./gradle_home`)

**Comment by monolied** [comment]
I was using `--no-build-cache` and thought that would avoid issues like that... Also In that case I'd be stumped by `--scan` *showing* the PKIX issues. Also, that wouldn't explain the build working when I use a sufficient keystore.

I gave it a try, however. When I set GRADLE_USER_HOME to an empty /tmp/gradletest after deleting ./build, I do see in fact a lot of output regarding certificates, but that is only about loading the handful of certificates from the keystore. After that, the failure message is like above, ie doesn't contain the certificate error.
So no; it seems to be pretty much independent of cache. Thank you for that proposal, however; it might be helpful in different circumstances.
(I was thinking if maybe the _somewhat_ exotic Temurin 17 JDK uses error messages for PKIX path building that don't get recognized by gradle, for whatever reason, and therefore are suppressed. Except, for other inexplicable reasons, when using `--scan`.)

**Comment by monolied** [comment]
Ah. My bad. When I changed GRADLE_USER_HOME to something else, the proxy configuration was gone, too. So later tests without build cache and empty home, but with using `-Djavax.net.debug=all` failed - albeit for an entirely different reason than before: without proxy gradle couldn't connect at all, and since the final error output is indistinguishable from the one before, I was led to believe nothing had changed. Only when I realised gradle was spending 3 minutes in `<======-------> 50% CONFIGURING [2m 59s]`, I thought something was amiss.

In fact, `-Djavax.net.debug=all` *does* show the `sun.security.validator.ValidatorException: PKIX path building failed`
*when* used with an appropriate handful of switches (`--no-daemon`, `--refresh-dependencies`, `--no-build-cache`).

 (`-Djava.security.debug=ssl` doesn't. )

So thanks a lot for your help!
